copy/copy.go

```diff
@@ -1159,7 +1159,7 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 			channel:  c.progress,
 			interval: c.progressInterval,
 			artifact: srcInfo,
-			lastTime: time.Now(),
+			lastTime: time.Time{},
```

Comment:
`copyBlobFromStream` should ideally not need to know about the `time.Time{}` special case; this rather calls for a `newProgressReader()` constructor that handles that.

- That constructor could legitimately fire a “dealing with new artifact, 0% progress” event, without making _this_ code any more complex.
- It would then be very reasonable to add a `defer func() { progressReader.reportDone(didReadAll, errorIfAny) }` or something like that (let’s not add data that no-one needs), and ensure that there really _is_ a 100%/skipped report.
- Perhaps we could also split the constructor from a `progressReader.assignSource(destStream)` call, and call the constructor earlier than here, to get the 0% indication in the right place.